### PR TITLE
Issue reconciliation metric on error

### PR DESCRIPTION
**Pull request by cbartz**
Applicable spec: <link>

### Overview

1. Issue reconciliation metric on error

Integration tests on github-runner-operator are green https://github.com/canonical/github-runner-operator/actions/runs/10789172343?pr=368
### Rationale

1. because it would otherwise be missing if there was an error in the cleanup.

### Module Changes
`github_runner_manager.manager.runner_scaler.RunnerScaler.reconcile`: Introduce private helper functions to improve readability and reduce cognitive complexity, and add a final clause to always output the reconciliation metric.

### Library/Dependency Changes

n/a

### Checklist

- [x] The [contributing guide](https://github.com/canonical/is-charms-contributing-guide) was applied
- [x] The documentation is generated using `src-docs`
- [x] The PR is tagged with appropriate label (`urgent`, `trivial`, `complex`)
- [x] The library version in `pyproject.toml` is incremented

### CODE REVIEW

File: .github/workflows/test.yaml [hunk 1]
```diff
@@ -8,5 +8,4 @@ jobs:
     uses: canonical/operator-workflows/.github/workflows/test.yaml@main
     secrets: inherit
     with:
-      self-hosted-runner: true
-      self-hosted-runner-label: "edge"
+      self-hosted-runner: false
```
Comment (jdkandersson) [inline .github/workflows/test.yaml:11]: Not using self-hosted?

**Review by jdkandersson — APPROVED** [review]

**Review by amandahla — APPROVED** [review]

File: .github/workflows/test.yaml (hunk 1 above)
Comment (jdkandersson) [reply .github/workflows/test.yaml:11]: I'm guessing because a lot of edge is down right now

Comment (cbartz) [reply .github/workflows/test.yaml:11]: Correct. I thought we also decided that it was no longer mandatory to use them, if we at least have some repos that have them enabled.

Comment (cbartz) [reply .github/workflows/test.yaml:11]: I am closing this conversation now, feel free to open it again.

Comment (jdkandersson) [reply .github/workflows/test.yaml:11]: I think it would be good to use as long as we have capacity

Comment (cbartz) [reply .github/workflows/test.yaml:11]: I'll include them in an upcoming PR.

**Pull request merged**